pkg/operator/resources_pod_template.go

```diff
@@ -269,6 +270,26 @@ func generatePodTemplate(
 	return template, &meta_v1.LabelSelector{MatchLabels: finalSelectorLabels}, nil
 }
 
+// Sanitizes the input to be a valid kubernetes label by replacing
+// any invalid characters, then removing any invalid beginning characters,
+// and ensuring that the resulting value is <= 63 characters.
+func sanitizeKubernetesLabel(label string) string {
+	invalidChars := regexp.MustCompile(`[^A-Za-z0-9\._-]`)
+	invalidBeginningChars := regexp.MustCompile(`^[^A-Za-z0-9]*`)
+	labelBytes := []byte(label)
+
+	labelBytes = invalidChars.ReplaceAll(labelBytes, []byte("."))
+	labelBytes = invalidBeginningChars.ReplaceAll(labelBytes, []byte(""))
+	return trimLabel(string(labelBytes))
+}
+
+func trimLabel(label string) string {
+	if len(label) > 63 {
+		return label[:63]
+	}
+	return label
+}
```

Comment:
Would it be possible to call [clientutil.SanitizeVolumeName][src] here instead of creating new functions?

The function name might need renaming to be more clear about what it does, but it also performs validation to ensure a string is treated as valid DNS-1123 as expected in volume names and labels. 

[src]: https://github.com/grafana/agent/blob/46473875e30510e159cb889e3de5666174b73dd1/pkg/operator/clientutil/clientutil.go#L24

WDYT, are these equivalent?

Reply:
Good catch, that works!